OpenRA.Game/Map/CellLayer.cs

```diff
@@ -18,13 +18,13 @@ namespace OpenRA
 	// Represents a layer of "something" that covers the map
 	public class CellLayer<T> : IEnumerable<T>
 	{
-		public readonly Size Size;
-		public readonly TileShape Shape;
+		public Size Size { get; private set; }
```

Comment:
Within OpenRA we have preferred `public readonly` fields because they provide a guarantee that the value will not change after initialization, and that it is therefore safe to cache any results derived from them.  It also helps prevents new bugs by providing a check against well-intentioned people replacing a value at runtime when outside code assumes it is constant.

These useful guarantees are lost if we start migrating to (private) settable properties, but if this is the only way we can efficiently move towards a testable system then I guess (but would like some stronger convincing) it is an acceptable tradeoff.

OpenRA differs from industry convention in many ways, but it is not exactly a conventional industry project.  There is usually logic behind these differences.